src/app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.h

```diff
@@ -600,6 +600,42 @@ class CameraAVStreamMgmtServer : public CommandHandlerInterface, public Attribut
         return CHIP_NO_ERROR;
     }
 
+    template <typename StreamContainer, typename IdGetter>
+    bool ValidateStreamForModifyOrDeallocateImpl(StreamContainer & streams, uint16_t streamID, HandlerContext & ctx,
+                                                 const char * streamTypeName, IdGetter id_getter)
+    {
+        auto it = std::find_if(streams.begin(), streams.end(), [&](const auto & stream) { return id_getter(stream) == streamID; });
+
+        if (it == streams.end())
+        {
+            ChipLogError(Zcl, "CameraAVStreamMgmt[ep=%d]: %s stream with ID: %u not found", mEndpointId, streamTypeName, streamID);
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::NotFound);
+            return false;
+        }
+
+        if (it->referenceCount > 0)
+        {
+            ChipLogError(Zcl, "CameraAVStreamMgmt[ep=%d]: %s stream with ID: %u still in use", mEndpointId, streamTypeName,
+                         streamID);
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::InvalidInState);
```

Comment:
Where does the spec say you can't modify a stream with nonzero refcount?

Reply:
Yes, the ref count checks are only performed on Stream Deallocation commands.